src/hermes/commands/init/base.py

```diff
@@ -0,0 +1,480 @@
+        self.folder_info = scout_current_folder()
+        sc.echo("Scan complete.", debug=True)
+
+    def __call__(self, args: argparse.Namespace) -> None:
+        # Test if init is possible and wanted. If not: sys.exit
+        self.test_initialization(args)
+
+        sc.echo(f"Starting to initialize HERMES in {self.folder_info.absolute_path} ...")
+        sc.max_steps = 6
+        sc.next_step("Configure deposition platform and setup method")
+
+        # Choosing desired deposit platform
+        self.choose_deposit_platform()
+
+        # Choosing setup method
+        self.setup_method = sc.choose(
+            f"How do you want to connect {DepositPlatformNames[self.deposit_platform]} "
+            f"with your {self.folder_info.used_git_hoster.name} CI?",
+            options={
+                "a": "automatically (using OAuth / Device Flow)",
+                "m": "manually (with instructions)"
+            },
+            default="a"
+        )
+
+        sc.next_step("Create hermes.toml file")
+
+        # Creating the hermes.toml file
+        self.create_hermes_toml()
+
+        sc.next_step("Create CITATION.cff file")
+
+        # Creating the citation File
+        self.create_citation_cff()
+
+        sc.next_step("Create git CI files")
+
+        # Adding .hermes to the .gitignore
+        self.update_gitignore()
+
+        # Creating the ci file
+        self.create_ci_template()
+
+        sc.next_step("Setup deposition platform")
+
+        # Connect with deposit platform
+        self.connect_deposit_platform()
+
+        sc.next_step("Setup git project")
+
+        # Adding the token to the git secrets & changing action workflow settings
+        self.configure_git_project()
+
+        sc.echo("")
+        sc.echo("HERMES is now initialized and ready to be used.", formatting=sc.Formats.OKGREEN+sc.Formats.BOLD)
+        sc.echo("")
+
+    def test_initialization(self, args: argparse.Namespace):
+        # Abort if git is not installed
+        if not is_git_installed():
+            sc.echo("Git is currently not installed. It is mandatory for HERMES to have git installed.")
+            sys.exit()
+
+        # Look at the current folder
+        self.refresh_folder_info()
+
+        # Abort if there is no git
+        if not self.folder_info.has_git:
+            sc.echo("The current directory has no `.git` subdirectory. "
+                    "Please execute `hermes init` in the root directory of your git project.",
+                    formatting=sc.Formats.FAIL)
+            sys.exit()
+
+        # Abort if neither GitHub nor gitlab is used
+        if self.folder_info.used_git_hoster == GitHoster.Empty:
+            sc.echo("Your git project ({}) is not connected to github or gitlab. It is mandatory for HERMES to "
+                    "use one of those hosting services.".format(self.folder_info.git_remote_url),
+                    formatting=sc.Formats.FAIL)
+            sys.exit()
```

Comment:
Good question. On the one side we could of course give someone with that setup a hermes.toml.
On the other side that is already all we can do; leaving the setup process at ~80% unfinished. The CI/CD stuff but also the Deposit Platform related stuff requires Git.
I fear that it might give a false sense of completeness or encourage them to try setting up Hermes with a platform that might not even be compatible.

Reply:
Why does the reposit stuff require GitLab or GitHub? 🤔 

Concerning the "completeness", we could print a message, something like:

>Skipped CI/CD setup (only available for GitLab/GitHub).

Maybe with a hint that they can write a feature request for their own Git host and/or CI platform.

Reply:
That sounds good. I will add it that way.

(For the Zenodo setup we need to save the access token as git project variable)

Reply:
Oh, yup that's right. But if there's no CI/CD the token is not needed. We could just tell the user to create one themselves (or create it for them and print it to the terminal, but I'm not sure about this).